src/lib.rs

```diff
@@ -166,25 +177,90 @@ pub enum PixelLayout {
     Unknown,
 }
 
+#[derive(Copy, Clone, Debug)]
+pub enum PlanarImageComponent {
+    Y,
+    U,
+    V,
+}
+
+impl std::convert::From<usize> for PlanarImageComponent {
+    fn from(index: usize) -> Self {
+        match index {
+            0 => PlanarImageComponent::Y,
+            1 => PlanarImageComponent::U,
+            2 => PlanarImageComponent::V,
+            _ => panic!("Invalid YUV index: {}", index),
+        }
+    }
+}
+
+impl std::convert::From<PlanarImageComponent> for usize {
+    fn from(component: PlanarImageComponent) -> Self {
+        match component {
+            PlanarImageComponent::Y => 0,
+            PlanarImageComponent::U => 1,
+            PlanarImageComponent::V => 2,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Plane(Picture, PlanarImageComponent);
+
+impl AsRef<[u8]> for Plane {
+    fn as_ref(&self) -> &[u8] {
+        let (width, height) = self.0.plane_data_geometry(self.1);
+        unsafe {
+            std::slice::from_raw_parts(
+                self.0.plane_data_ptr(self.1) as *const u8,
+                (width * height) as usize,
+            )
+        }
+    }
+}
+unsafe impl Send for Plane {}
+unsafe impl Sync for Plane {}
+
 #[derive(Copy, Clone, Debug)]
 pub struct BitsPerComponent(pub usize);
 
 impl Picture {
-    pub fn stride(&self, component: usize) -> i32 {
-        (*self.pic).stride[component] as i32
+    pub fn stride(&self, component: PlanarImageComponent) -> i32 {
+        let s = match component {
+            PlanarImageComponent::Y => 0,
+            _ => 1,
+        };
+        (*self.inner).pic.stride[s] as i32
+    }
+
+    pub fn plane_data_ptr(&self, component: PlanarImageComponent) -> *mut c_void {
+        let index: usize = component.into();
+        (*self.inner).pic.data[index]
+    }
+
+    pub fn plane_data_geometry(&self, component: PlanarImageComponent) -> (u32, u32) {
+        let height = match component {
+            PlanarImageComponent::Y => self.height(),
+            _ => {
+                let ss_ver = (self.pixel_layout() == PixelLayout::I420) as u32;
```

Comment:
Instead of this and the line below, maybe a match on the pixel layout to do it explicitly would be better? That way it won't become wrong once a new pixel layout is added and to make things compile a new case would have to be added to the match.

Also casting a `bool` to a `u32` is a bit weird :)

Reply:
Sure, I can make it a match. About the cast, do you have a better suggestion?

Reply:
```
match self.pixel_layout() {
    PixelLayout::I420 => (self.height() + 1) / 2,
    PixelLayout::I444 | PixelLayout::I422 | etc => self.height(),
}